python/spinn/util/chainer_blocks.py

```diff
@@ -226,6 +275,31 @@ def slice_gate(gate_data, i):
     return (c_t, h_t)
 
 
+def treegru(h_left, h_right, gates, U_left, U_right):
```

Comment:
Should we add assertions for this ? Apart from preventing bad usage, it would also act as a comment.